lib/HLSL/HLMatrixLowerPass.cpp

```diff
@@ -2330,9 +2330,66 @@ void HLMatrixLowerPass::DeleteDeadInsts() {
   m_inDeadInstsSet.clear();
 }
 
+// Iterate GV users, and transparently iterate address space cast users as well
+class GVUserIter {
```

Comment:
In the end, this doesn't have anything to do with global variables, except that that's where it's used? You could call it `AddrSpaceAgnosticUserIter` or whatever.

Reply:
It really only makes sense for GV, since it only tunnels through constant address space cast, and skips unused constant users.  It's for one to iterate users of the GV, without having to add recursion in order to deal with this one address space cast case.  I could rename it, but not sure the suggested is complete either, since it skips unused constant users as well.  I'll name it the very long, but descriptive: `UserIter_TunnelAddrSpace_SkipUnusedConstantUser`.